### PR TITLE
Unify hardware feature detection between CoreCLR JIT and AOT

### PR DESCRIPTION
**Pull request by jkotas**
Follow up on #87865

### CODE REVIEW

File: src/coreclr/gc/vxsort/isa_detection.cpp [hunk 1]
```diff
@@ -19,28 +19,6 @@ enum class SupportedISA
...
-{
-    // On Windows we have an api(GetEnabledXStateFeatures) to check if AVX is supported
-    typedef DWORD64(WINAPI* PGETENABLEDXSTATEFEATURES)();
-    PGETENABLEDXSTATEFEATURES pfnGetEnabledXStateFeatures = NULL;
-
-    HMODULE hMod = LoadLibraryExW(L"kernel32.dll", NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
-    if (hMod == NULL)
-        return 0;
-
-    pfnGetEnabledXStateFeatures = (PGETENABLEDXSTATEFEATURES)GetProcAddress(hMod, "GetEnabledXStateFeatures");
```
Comment (jkotas) [inline src/coreclr/gc/vxsort/isa_detection.cpp:32]: https://github.com/dotnet/runtime/blob/08a6e06d8fdc29efebc5c0ec7d2b6ba215810e71/src/coreclr/vm/threadsuspend.cpp#L1972 links to `GetEnabledXStateFeatures` directly, so it is safe to delete the dynamic linking here.

File: src/coreclr/nativeaot/Runtime/i386/MemClrForGC.asm [hunk 2]
```diff
@@ -1,148 +0,0 @@
-;; Licensed to the .NET Foundation under one or more agreements.
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/i386/MemClrForGC.asm:1]: I have noticed these unused files thanks to `IsProcessorFeaturePresent` call.

**Comment by am11** [comment]
cc @gbalykov, if we ever added RISC-V support in `src/native/minipal/cpufeatures.c`, I think we can make it return the features per-hart (as each hardware thread in RISC-V can technically have different features, e.g. [virtual memory range sizes](https://github.com/dotnet/runtime/commit/82640bd2f4cf5b9e8e38f2384ecb7d6cb309a6a9); for which GC can be fine-tuned).

File: src/coreclr/gc/vxsort/isa_detection.cpp (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/gc/vxsort/isa_detection.cpp:32]: It's worth noting that the original reason for the dynamic linking was to support Windows 7 prior to SP1.

With Win7 support having been dropped and us just not regressing Win 7 SP1 and later, this is fine.

File: src/coreclr/vm/codeman.cpp [hunk 3]
```diff
@@ -1338,531 +1277,262 @@ void EEJitManager::SetCpuInfo()
...
     {
-        CPUCompileFlags.Clear(InstructionSet_SSE3);
+        CPUCompileFlags.Set(InstructionSet_X86Serialize);
     }
+#elif defined(TARGET_ARM64)
 
-    if (!CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableSSE41))
+#if !defined(TARGET_WINDOWS)
+    // Linux may still support no AdvSimd
+    if ((cpuFeatures & InstructionSet_AdvSimd) == 0)
```
Comment (tannergooding) [inline src/coreclr/vm/codeman.cpp:1461]: Should this just check for `InstructionSet_VectorT128` like the xarch path does?

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/vm/codeman.cpp (hunk 3 above)
Comment (jkotas) [reply src/coreclr/vm/codeman.cpp]: Makes sense.

**Pull request merged**